src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -960,11 +960,10 @@ private async Task<WorkUnitResult> ExecuteInstantiatedTask(ITaskExecutionHost ta
                     && !taskResult // and it returned false
                     && !taskLoggingContext.HasLoggedErrors // and it didn't log any errors
                     && (be is TaskHost th ? th.BuildRequestsSucceeded : false)
-                    && (be is IBuildEngine7 be7 ? !be7.AllowFailureWithoutError : true) // and it's not allowed to fail unless it logs an error
                     && !(_cancellationToken.CanBeCanceled && _cancellationToken.IsCancellationRequested)) // and it wasn't cancelled
                 {
                     // Then decide how to log MSB4181
-                    if (_continueOnError == ContinueOnError.WarnAndContinue)
+                    if (_continueOnError == ContinueOnError.WarnAndContinue || (be is IBuildEngine7 be7 && be7.AllowFailureWithoutError))
```

Comment:
Team triage: We think the AllowFailureWithoutError half of this should log a message instead of a warning. (ContinueOnError should still be a warning, of course.) Thanks!

Reply:
@FrtLoop, did you see our suggestion?

Reply:
@FrtLoop, are you planning to get back to this soon? Otherwise, I might close it until you're ready.